### ISSUE TITLE
Confusion regarding Freemode Female texture editing via OpenIV

### ISSUE THREAD

**Issue by TerraXanthis** [gtav-ent/GTAV-EnhancedNativeTrainer#96]
This may be completely in the wrong place, but I'm using Enhanced Native Trainer and I was hoping you might help me glean some insight into this issue.

In an attempt to play as my GTA Online character in GTA V Story Mode, I'm using the Enhanced Native Trainer skin changer to switch to Freemode Female and recreating my Online character. However, this trainer, and every other skin editor I've found, lacks the ability to customize hair color and makeup. My character just doesn't look quite the same without the makeup, so as a temporary workaround, I decided to manually edit the textures via OpenIV to look similar, if not exactly the same.

My main issue now is that the changes to the face textures aren't showing up in-game. What I did was I navigated to x64v.rpf/models/cdimages/streamedpeds_mp.rpf/mp_f_freemode_01 and exported hair_diff_004_f_uni.ytd (bluish grey ponytail texture) and head_diff_021_a_whi.ytd (Female head 21 texture). I opened up the face texture in photoshop and overlaid mp_fm_faov_makeup_002.ytd (bluish eyeshadow) from ped_mp_overlay_txds.rpf. Saved them both as .dds files then imported them back into OpenIV and replaced the old ones.

Now, for whatever reason, when I open head_021_r.ydd with OpenIV's model viewer, I see the changes I made to the face applied perfectly. However, the new texture simply isn't applying in-game.

This is the confusing bit. The changes I made to the hair color worked perfectly. I replaced hair_diff_004_f_uni.ytd with my new color and it shows up in-game just fine. It's just the face texture that's not working.

I've tried several other faces, and done several different obvious edits, but I simply can't seem to actually edit the Online character faces.

As I said, any insight would be appreciated. Am I simply looking in the wrong directory-- does Enhanced Native Trainer specifically take the head models from a different folder? Better yet-- any plans for an on-the-fly makeup/hair color menu in-game? (It'd certainly simplify things, I'll say that much.)

Thanks in advance, and best of luck on future updates to the trainer!

**Comment by gtav-ent** [comment]
The MP characters have blended faces based on two "parent" faces, don't they? We don't support this, not least because I can't figure out how it works.

**Comment by TerraXanthis** [comment]
That's a good point, actually. The models in mp_f_freemode_01 I'm looking at might just be default combinations of the parent faces. Though that still doesn't explain why the texture edits are showing up fine in OpenIV and not at all in-game. Nevertheless, I suppose I'll keep messing with things to try and figure this problem out myself, in due time. Thanks for the response, at the very least!

**Comment by gtav-ent** [comment]
No worries. If you find anything out, be it new research or using someone else's stuff, please let us know here.

**Comment by TerraXanthis** [comment]
PROBLEM SOLVED!

Phew.

As it turns out, the problem was excruciatingly simple. It turns out, specifically the face textures of the MP characters, as well as a few odd hair textures, plus some other miscellaneous files for other characters, were updated/altered and replaced in a patch. This is simply inferring what we know is true: anything polaced in the update folder/.rpf replaces any other instance of it automatically. So, the files in x64v.rpf were being overwritten by these new patched textures.

For posterity, the directory I found these updated folders in is:
update/x64/dlcpacks/mppatchesng/dlc.rpf/x64/models/cdimages/mppatches.rpf

I simply replaced the patched head_diff_021_a_whi.ytd with my edited version, and it's working with absolutely no problems in-game. I hope this information might prove useful to you!
